pkg/sdkclient/serverinfo/serverinfo.go

```diff
@@ -63,20 +64,32 @@ func waitForServerInfo(timeout time.Duration, filePath string) (*info.ServerInfo
 	}
 
 	sdkVersion := serverInfo.Version
+	minNumaflowVersion := serverInfo.MinimumNumaflowVersion
+	sdkLanguage := serverInfo.Language
 	numaflowVersion := numaflow.GetVersion().Version
 
-	// If we are testing locally or in CI, we can skip checking for numaflow compatibility issues
-	// because both return us a version string that the version check libraries can't properly parse. (local: "*latest*" CI: commit SHA)
-	if !strings.Contains(numaflowVersion, "latest") && !strings.Contains(numaflowVersion, numaflow.GetVersion().GitCommit) {
-		if err := checkNumaflowCompatibility(numaflowVersion, serverInfo.MinimumNumaflowVersion); err != nil {
+	// If MinimumNumaflowVersion is empty, skip the numaflow compatibility check as there was an
+	// error writing server info on the SDK side
+	if minNumaflowVersion == "" {
+		log.Printf("warning: failed to get minimum numaflow version")
+		// If we are testing locally or in CI, we can skip checking for numaflow compatibility issues
+		//because both return us a version string that the version check libraries can't properly parse (local: "*latest*" CI: commit SHA)
```

Comment:
```suggestion
		// because both return us a version string that the version check libraries can't properly parse (local: "*latest*" CI: commit SHA)
```